### PR TITLE
Refactor/split outbox and subscriber

### PR DESCRIPTION
**Pull request by lzukowski**

### CODE REVIEW

File: event_sourcery_sqlalchemy/__init__.py [hunk 1]
```diff
@@ -3,29 +3,86 @@
...
+    def subscribe_in_transaction(self) -> InTransactionSubscription:
+        return InTransactionSubscription(self.serde)
 
 
+@dataclass(repr=False)
 class SQLStoreFactory(EventStoreFactory):
-    def __init__(self, session: Session) -> None:
-        self._session = session
-        self._configure(storage_strategy=SqlAlchemyStorageStrategy(self._session))
+    session: Session
```
Comment (Enforcer) [inline event_sourcery_sqlalchemy/__init__.py:48]: Does it have to be public?

File: event_sourcery/event_store/factory.py [hunk 2]
```diff
@@ -1,49 +1,47 @@
...
+    outbox: Outbox
+    subscriber: subscription.Positioner
 
-    def _configure(self, **kwargs: Any) -> Self:
-        self.build = partial(self.build, **kwargs)
-        return self
 
+class EventStoreFactory(abc.ABC):
+    @abc.abstractmethod
+    def build(self) -> Engine:
```
Comment (Enforcer) [inline event_sourcery/event_store/factory.py:34]: I'm not feeling it fully. Why not have separate methods for `build_event_store`, `build_outbox` etc?

File: event_sourcery/event_store/factory.py [hunk 3]
```diff
@@ -1,49 +1,47 @@
...
-    build: Callable[..., EventStore] = partial(
-        EventStore,
-        outbox_storage_strategy=NoOutboxStorageStrategy(),
-        serde=serde,
-    )
+class Engine:
+    serde: Serde
+    event_store: EventStore
+    outbox: Outbox
+    subscriber: subscription.Positioner
```
Comment (Enforcer) [inline event_sourcery/event_store/factory.py:29]: What's `Positioner` actually?

File: event_sourcery/event_store/factory.py (hunk 2 above)
Comment (lzukowski) [reply event_sourcery/event_store/factory.py:34]: It's like committing to configuration.

For some reason I want it to be resilient for edge cases like building an event store and changing config, building outbox.

File: event_sourcery/event_store/factory.py (hunk 3 above)
Comment (lzukowski) [reply event_sourcery/event_store/factory.py:29]: It's an entry for subscription. First, you need to choose a position to start from. Then API forces you to choose the type of subscription, and then you can build a batch or iterator.

File: event_sourcery_sqlalchemy/__init__.py (hunk 1 above)
Comment (lzukowski) [reply event_sourcery_sqlalchemy/__init__.py:48]: No, changing.

**Pull request merged**